lang/funcs/core/net/mac_fmt_func.go

```diff
@@ -0,0 +1,50 @@
+// Mgmt
+// Copyright (C) 2013-2020+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package corenet
+
+import (
+	"net"
+	"strings"
+
+	"github.com/purpleidea/mgmt/lang/funcs/simple"
+	"github.com/purpleidea/mgmt/lang/types"
+)
+
+func init() {
+	simple.ModuleRegister(ModuleName, "macfmt", &types.FuncValue{
+		T: types.NewType("func(a str) str"),
+		V: MacFmt,
+	})
+}
+
+// MacFmt takes a mac address with hyphens and converts it to a format with
+// colons.
+func MacFmt(input []types.Value) (types.Value, error) {
+	mac := input[0].Str()
+
+	// Check if the MAC is valid
+	_, err := net.ParseMAC(mac)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Should we check anything else for sanity? Ensure the length is correct for example? Specify why or why not, there is a specific reason, and add it to the code comments.

Reply:
There is `net.ParseMac()` that check if we have a valid MAC. If not, it returns an error. However I am not sure if it's relevant to add another validation on top of that unless we don't want anything longer than a EUI-48 MAC ?

Reply:
Bingo-- the broad definition of a mac address that the upstream golang lib uses is probably not what we want or need in our version.